src/jit/importer.cpp

```diff
@@ -3607,7 +3614,33 @@ GenTreePtr Compiler::impIntrinsic(CORINFO_CLASS_HANDLE  clsHnd,
             retNode = op1;
             break;
 #endif
-
+        // Implement ByReference Ctor.  This wraps the assignment of the ref to the span field.
+        // In effect this is just a substitution.  The parameter byref will be substituted for the
+        // call node.
+        case CORINFO_INTRINSIC_ByReference_Ctor:
+        {
+            // Remove call to constructor and directly assign the byref passed
+            // to the call to the first slot of the ByReference struct.
```

Comment:
Is this going to scale once we create other ByRef types with the byref pointer not being the first slot in the CLR type? e.g. a struct with the first field of type int followed by span. Will such struct be considered ByReference type by JIt or it will be just a struct and the "ByReference" logic will kick in when the span field's ctor is called?